cryptoutil/ecdsa.go

```diff
@@ -40,8 +40,8 @@ func (s *ECDSASigner) KeyID() (string, error) {
 	return GeneratePublicKeyID(&s.priv.PublicKey, s.hash)
 }
 
-func (s *ECDSASigner) Sign(r io.Reader) ([]byte, error) {
-	digest, err := Digest(r, s.hash)
+func (s *ECDSASigner) Sign(ctx context.Context, data []byte) ([]byte, error) {
```

Comment:
does this need to be depricated?

Reply:
I presume you mean the switch from taking in an `io.Reader` vs `Context` and `[]byte`?

So this hasn't been documented at all yet, and probably needs doing so, but switching to this is to align closer to the [secure systems lab lib](https://github.com/secure-systems-lab/dsse), so moving over to it is less hassle when we do.

However, passing in an `io.Reader` might be more beneficial for making all of this test driven, so it probably warrants a wider discussion. Also there might be other perfectly good reasons for this being passed as an argument.

To answer the original question, I am not sure about the need for deprecation inside this `cryptoutil` library more generally. Are we intending for it to be used directly? I have also been thinking about embedding it within the `signature` package, and only exposing it in the form of `NewSignerFromReader`/`NewSignerFromFile`. Not sure what your thoughts are on this?